ReactWindows/ReactNative.Net46/ReactPage.cs

```diff
@@ -15,21 +14,35 @@ namespace ReactNative
     /// </summary>
     public abstract class ReactPage : Page, IAsyncDisposable
     {
-        private readonly IReactInstanceManager _reactInstanceManager;
-
-        private bool _isShiftKeyDown;
-        private bool _isControlKeyDown;
+        private readonly Lazy<IReactInstanceManager> _reactInstanceManager;
+        private readonly Lazy<ReactRootView> _rootView;
 
         /// <summary>
         /// Instantiates the <see cref="ReactPage"/>.
         /// </summary>
         protected ReactPage()
         {
-            _reactInstanceManager = CreateReactInstanceManager();
-            RootView = CreateRootView();
-            Content = RootView;
+            this._reactInstanceManager = new Lazy<IReactInstanceManager>(() =>
+            {
+                DispatcherHelpers.CurrentDispatcher = this.Dispatcher;
+
+                var reactInstanceManager = CreateReactInstanceManager();
+
+                return reactInstanceManager;
+            });
+
+            this._rootView = new Lazy<ReactRootView>(() =>
```

Comment:
I am going to remove all this. except the one where I'm referencing dispatcher from the Page base class where I'll use base.Dispatcher to make it clear it is using a class member not from the immediate class..